lib/extensions/ar_virtual.rb

```diff
@@ -66,43 +66,99 @@ module ClassMethods
     #
 
     def virtual_delegate(*methods)
-      options = methods.pop
-      unless options.kind_of?(Hash) && options[:to]
+      options = methods.extract_options! || {}
+      unless (to = options[:to])
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
-      delegate(*methods, options)
+
+      to = to.to_s
+      if to.include?(".") && methods.size > 1
+        raise ArgumentError, 'Delegation only supports specifying a method name when defining a single virtual method'
+      end
+
+      location = caller_locations(1, 1).first
+      file, line = location.path, location.lineno
```

Comment:
These two lines are here only to be used by the `define_delgate` method.  Would it be possible to just call these within the `define_delgate` method itself?  You might have to update the first line if you want to provide an accurate location for errors, because you would now be farther down the stack:

```ruby
location = caller_locations(2,1).first
```

https://ruby-doc.org/core-2.1.0/Module.html#method-i-module_eval
https://ruby-doc.org/core-2.1.0/Kernel.html#method-i-caller_locations